infrastructures/infrastructure-gce/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/GCEInfrastructure.java

```diff
@@ -0,0 +1,361 @@
+    protected String vmPrivateKey = null;
+
+    @Configurable(description = "Resource manager hostname or ip address (must be accessible from nodes)")
+    protected String rmHostname = generateDefaultRMHostname();
+
+    @Configurable(description = "Connector-iaas URL")
+    protected String connectorIaasURL = "http://" + generateDefaultRMHostname() + ":8080/connector-iaas";
+
+    @Configurable(description = "Command used to download the node jar")
+    protected String downloadCommand = linuxInitScriptGenerator.generateNodeDownloadCommand(rmHostname);
+
+    @Configurable(description = "(optional) Additional Java command properties (e.g. \"-Dpropertyname=propertyvalue\")")
+    protected String additionalProperties = "-Dproactive.useIPaddress=true";
+
+    @Configurable(description = "(optional) The image of the virtual machine")
+    protected String image = DEFAULT_IMAGE;
+
+    @Configurable(description = "(optional) The region of the virtual machine")
+    protected String region = DEFAULT_REGION;
+
+    @Configurable(description = "(optional) The minimum RAM required (in Mega Bytes) for each virtual machine")
+    protected int ram = DEFAULT_RAM;
+
+    @Configurable(description = "(optional) The minimum number of CPU cores required for each virtual machine")
+    protected int cores = DEFAULT_CORES;
+
+    @Override
+    public void configure(Object... parameters) {
+        logger.info("Validating parameters : " + Arrays.toString(parameters));
+        validate(parameters);
+
+        int parameterIndex = 0;
+
+        this.gceCredential = getCredentialFromJsonKeyFile((byte[]) parameters[parameterIndex++]);
+        this.numberOfInstances = Integer.parseInt(parameters[parameterIndex++].toString().trim());
+        this.numberOfNodesPerInstance = Integer.parseInt(parameters[parameterIndex++].toString().trim());
+        this.vmUsername = parameters[parameterIndex++].toString().trim();
+        this.vmPublicKey = new String((byte[]) parameters[parameterIndex++]);
+        this.vmPrivateKey = new String((byte[]) parameters[parameterIndex++]);
+        this.rmHostname = parameters[parameterIndex++].toString().trim();
+        this.connectorIaasURL = parameters[parameterIndex++].toString().trim();
+        this.downloadCommand = parameters[parameterIndex++].toString().trim();
+        this.additionalProperties = parameters[parameterIndex++].toString().trim();
+        this.image = parameters[parameterIndex++].toString().trim();
+        this.region = parameters[parameterIndex++].toString().trim();
+        this.ram = Integer.parseInt(parameters[parameterIndex++].toString().trim());
+        this.cores = Integer.parseInt(parameters[parameterIndex++].toString().trim());
+
+        connectorIaasController = new ConnectorIaasController(connectorIaasURL, INFRASTRUCTURE_TYPE);
+    }
+
+    private void validate(Object[] parameters) {
+        if (parameters == null || parameters.length < 14) {
```

Comment:
Replace `14` by a variable or attribute. Here it can be considered as a magic number.

Reply:
Modified. Thanks.